COMPILING.md

```diff
@@ -104,7 +104,9 @@ builds.
 
 - `-DCMAKE_BUILD_TYPE=RelWithDebInfo`: build a release build. This is
    significantly faster than a debug build, but has far less useful information
-   in case of a crash.
+   in case of a crash. Binaries still contain symbols, aka "unstripped".
+- `-DCMAKE_BUILD_TYPE=Release`: Same as previous option, without symbols,
+   aka "stripped".
```

Comment:
The target group of this document is people that want to self-compile OpenTTD. In most of these cases, `RelWithDebInfo` really is the better choice. `Release` is only useful if you actually want to publish your release, for which this document is not a good fit.

I am  not in favour of giving people options in documents like this, as it requires them to understand context. What are "symbols"? What is "stripped"? Most people wouldn't know nor care. The original text, most likely intentional, stayed away from these wording, and pointed out what the result was: faster, but less information in case of a crash.

Even so, `RelWithDebInfo` really is the better pick. In short: I would suggest that we don't make this change. Curious about other developer's opinions.

Reply:
I tend to agree, mentioning `Release` here is not helpful, it's part of the "known" cmake options.
We mention `RelWithDebInfo` because it makes sense as default value is `Debug` and is dead slow.

Reply:
I also agree...

Reply:
> I tend to agree, mentioning `Release` here is not helpful, it's part of the "known" cmake options. We mention `RelWithDebInfo` because it makes sense as default value is `Debug` and is dead slow.

Based on the "well-known" logic, it makes no sense to mention `RelWithDebInfo` either, as it is mentioned in [CMake documentation](https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html).